script/transform-approved-books-data.ts

```diff
@@ -0,0 +1,168 @@
+}
+
+const isApprovedVersion = (something: any): something is ApprovedVersion => {
+  return typeof something.content_version === 'string' && typeof something.min_code_version === 'string';
+};
+
+interface ApprovedVersionCollection extends ApprovedVersion {
+  collection_id: string;
+}
+
+const isApprovedVersionCollection = (something: any): something is ApprovedVersionCollection => {
+  return typeof something.collection_id === 'string' && isApprovedVersion(something);
+};
+
+interface ApprovedVersionRepo extends ApprovedVersion {
+  repo: string;
+}
+
+const isApprovedVersionRepo = (something: any): something is ApprovedVersionRepo => {
+  return typeof something.repo === 'string' && isApprovedVersion(something);
+};
+
+interface ApprovedBooksAndVersions {
+  approved_books: Array<ApprovedCollection | ApprovedRepo>;
+  approved_versions: Array<ApprovedVersionCollection | ApprovedVersionRepo>;
+}
+
+const isApprovedBooksAndVersions = (something: any): something is ApprovedBooksAndVersions => {
+  return Array.isArray(something.approved_books)
+    && something.approved_books.every(
+      (element: any) => isApprovedCollection(element) || isApprovedRepo(element))
+    && something.approved_versions.every(
+      (element: any) => isApprovedVersionCollection(element) || isApprovedVersionRepo(element));
+};
+
+const matchRepoVersion = (repoData: ApprovedRepo) =>
+  (versionData: ApprovedVersionCollection | ApprovedVersionRepo): versionData is ApprovedVersionRepo => {
+  return isApprovedVersionRepo(versionData) && versionData.repo === repoData.repo;
+};
+
+const matchCollectionVersion = (collectionData: ApprovedCollection) =>
+  (versionData: ApprovedVersionCollection | ApprovedVersionRepo): versionData is ApprovedVersionCollection => {
+  return isApprovedVersionCollection(versionData) && versionData.collection_id === collectionData.collection_id;
+};
+
+const getDesiredVersion = (
+  approvedVersions: Array<ApprovedVersionCollection | ApprovedVersionRepo>,
+  colOrRepo: ApprovedRepo | ApprovedCollection
+): string | undefined => {
+  const isCollection = isApprovedCollection(colOrRepo);
+
+  const filter = isCollection
+    ? matchCollectionVersion(colOrRepo as ApprovedCollection)
+    : matchRepoVersion(colOrRepo as ApprovedRepo);
+
+  const transformVersion = (versionData: ApprovedVersionCollection | ApprovedVersionRepo) => {
+    // books from collections have versions starting with 1.
+    return isCollection ? versionData.content_version.slice(2) : versionData.content_version;
+  };
+
+  const versions = approvedVersions
+    // only versions for current repo / collection
+    .filter(filter)
+    .map(transformVersion)
+    // sort and revert so results are descending
+    .sort()
+    .reverse();
+
+  // in general we want the most recent one, in the future we'll have to filter this by which ones
+  // are available for the recipe code version REX supports
+  return versions[0];
+};
+
+const { data } = argv as { data?: string };
```

Comment:
I think the script arguments should be read inside the main `transformData` function or maybe make the `transformData` function to receive the `data` as argument. 

Doesn't change anything but seems clearer to me ;)